### PR TITLE
expression: add builtin_time - hour/minute/second/micro_second and improve tests

### PR DESCRIPTION
**Pull request by koushiro**
## What have you changed? (mandatory)

- add builtin_time functions - hour/minute/second/micro_second
- improve tests

## What are the type of the changes? (mandatory)

- Improvement 

## How has this PR been tested? (mandatory)

unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

## Refer to a related PR or issue link (optional)

[#3275](https://github.com/tikv/tikv/issues/3275)

### CODE REVIEW

File: src/coprocessor/dag/expr/builtin_time.rs [hunk 1]
```diff
@@ -148,17 +172,75 @@ impl ScalarFunc {
...
+        let f = scalar_func_expr(sig, &[arg]);
+        let op = Expression::build(ctx, f).unwrap();
+        let got = op.eval(ctx, &[]);
+        match got {
+            Ok(got) => assert_eq!(got, exp),
+            Err(_) => assert!(true),
+        }
+    }
+
+    fn test_ok_case_two_arg(
```
Comment (ice1000) [inline src/coprocessor/dag/expr/builtin_time.rs]: Missing `#[test]`

File: src/coprocessor/dag/expr/builtin_time.rs [hunk 2]
```diff
@@ -148,17 +172,75 @@ impl ScalarFunc {
...
+        let f = scalar_func_expr(sig, &[arg]);
+        let op = Expression::build(ctx, f).unwrap();
+        let got = op.eval(ctx, &[]);
+        match got {
+            Ok(got) => assert_eq!(got, exp),
+            Err(_) => assert!(false, "eval failed"),
+        }
+    }
+
+    fn test_err_case_one_arg(ctx: &mut EvalContext, sig: ScalarFuncSig, arg: Datum, exp: Datum) {
```
Comment (ice1000) [inline src/coprocessor/dag/expr/builtin_time.rs]: Missing `#[test]`

File: src/coprocessor/dag/expr/builtin_time.rs [hunk 3]
```diff
@@ -148,17 +172,75 @@ impl ScalarFunc {
...
-    use coprocessor::codec::mysql::Time;
+    use coprocessor::codec::mysql::{Duration, Time};
     use coprocessor::codec::Datum;
     use coprocessor::dag::expr::tests::{datum_expr, scalar_func_expr};
     use coprocessor::dag::expr::*;
     use coprocessor::dag::expr::{EvalContext, Expression};
     use std::sync::Arc;
     use tipb::expression::ScalarFuncSig;
 
+    fn test_ok_case_one_arg(ctx: &mut EvalContext, sig: ScalarFuncSig, arg: Datum, exp: Datum) {
```
Comment (ice1000) [inline src/coprocessor/dag/expr/builtin_time.rs]: Missing `#[test]`

File: src/coprocessor/dag/expr/builtin_time.rs (hunk 1 above)
Comment (koushiro) [reply src/coprocessor/dag/expr/builtin_time.rs]: Functions used as tests can not have any arguments.
These are just functions to help tests.

Comment (ice1000) [reply src/coprocessor/dag/expr/builtin_time.rs]: Oh, sorry.

**Pull request merged**